RealmBrowserSync/Controllers/RLMSyncServerBrowserWindowController.m

```diff
@@ -79,27 +77,28 @@ - (void)showWindow:(id)sender {
 
     NSURL *adminRealmURL = [self.serverURL URLByAppendingPathComponent:RLMAdminRealmServerPath];
 
-    self.schemaLoader = [[RLMDynamicSchemaLoader alloc] initWithSyncURL:adminRealmURL user:self.user];
+    RLMRealmConfiguration *configuration = [[RLMRealmConfiguration alloc] init];
+    configuration.dynamic = YES;
+    configuration.syncConfiguration = [[RLMSyncConfiguration alloc] initWithUser:self.user realmURL:adminRealmURL];
 
     __weak typeof(self) weakSelf = self;
-    [self.schemaLoader loadSchemaWithCompletionHandler:^(NSError *error) {
-        if (error != nil) {
+    [RLMRealm asyncOpenWithConfiguration:configuration callbackQueue:dispatch_get_main_queue() callback:^(RLMRealm *realm, NSError *error) {
+        if (!weakSelf || !weakSelf.window.isVisible) {
+            // A window has been closed before realm is loaded
+            return;
+        }else if (error) {
             [[NSAlert alertWithError:error] beginSheetModalForWindow:weakSelf.window completionHandler:^(NSModalResponse returnCode) {
                 [weakSelf close];
             }];
         } else {
-            [weakSelf openAdminRealmAtURL:adminRealmURL withUser:weakSelf.user];
+            [weakSelf openAdminRealmWithConfiguration:configuration];
```

Comment:
The name `openAdminRealmWithConfiguration:` seems a little confusing now, since we've just opened the admin Realm, and have it available as `realm`. Perhaps this should be changed to `didOpenAdminRealm:` and take the `RLMRealm` as an argument?